bundles/framework/timeseries/view/TimeseriesControlPlugin.js

```diff
@@ -0,0 +1,642 @@
+                '<div class="timeseries-back"></div><div class="timeseries-playpause"></div><div class="timeseries-forward"></div><div class="timeseries-datetime"></div>' +
+                '</div>');
+            var dateTime = template.find('.timeseries-datetime');
+            me._updateTimeDisplay = function () {
+                dateTime.text(me.loc('dateRender', { val: new Date(me._uiState.currentTime) }));
+            };
+            template.find('.timeseries-playpause').on('click', function (e) {
+                me._setAnimationState(!me._uiState.isAnimating);
+            });
+            template.find('.timeseries-back').on('click', this._doSingleStep.bind(this, -1));
+            template.find('.timeseries-forward').on('click', this._doSingleStep.bind(this, 1));
+            this._element.find('.timeseries-aux').append(template);
+        },
+        /**
+         * @method _updateCurrentTime Update current time in response to user input
+         * @private
+         * @param  {String} newTime time instant, ISO-string
+         */
+        _updateCurrentTime: function (newTime) {
+            newTime = this._getClosestTime(newTime);
+            if (this._uiState.currentTime !== newTime) {
+                this._uiState.currentTime = newTime;
+                this._throttleNewTime();
+            }
+            this._renderHandle();
+            this._updateTimeDisplay();
+        },
+        /**
+         * @method _getTickFormatter Get formatting function for current locale
+         * @private
+         * @return {Function} time formatting function
+         */
+        _getTickFormatter: function () {
+            var localeString = Oskari.getLang();
+            var locale;
+            var formatterFunction;
+            if (this.__localeData[localeString]) {
+                locale = d3.timeFormatLocale(this.__localeData[localeString]);
+                formatterFunction = locale.format.bind(locale);
+            } else {
+                formatterFunction = d3.timeFormat.bind(d3);
+            }
+            var formatMillisecond = formatterFunction(".%L"),
+                formatSecond = formatterFunction(":%S"),
+                formatMinute = formatterFunction(locale ? "%H:%M" : "%I:%M"),
+                formatHour = formatterFunction(locale ? "%H:%M" : "%I %p"),
+                formatDay = formatterFunction(locale ? "%d.%m." : "%d %b"),
+                formatMonth = formatterFunction("%b"),
+                formatYear = formatterFunction("%Y");
+
+            return function multiFormat(date) {
+                return (d3.timeSecond(date) < date ? formatMillisecond
+                    : d3.timeMinute(date) < date ? formatSecond
+                    : d3.timeHour(date) < date ? formatMinute
+                    : d3.timeDay(date) < date ? formatHour
+                    : d3.timeMonth(date) < date ? formatDay
+                    : d3.timeYear(date) < date ? formatMonth
```

Comment:
We are using both moment.js and d3 for date/time operations?

Reply:
I guess it's ok to use d3 here as well since there's other d3 stuff in this particular file.

Reply:
moment.js is used for time duration/interval etc. handling, d3 for time format presentation.